### ISSUE TITLE
has no exported member 'Tapable'. error appears when trying to build a TypeScript project

### ISSUE THREAD

**Issue by wujekbogdan** [webpack/tapable#110]
I'm getting the following errors when trying to build a TypeScript project:

```
> functions@ build /Users/wujekbogdan/htdocs/rss-parser/functions
> tsc

../../../node_modules/@types/webpack/index.d.ts:25:10 - error TS2305: Module '"/Users/wujekbogdan/htdocs/node_modules/@types/tapable/index"' has no exported member 'Tapable'.

25 import { Tapable, HookMap,
            ~~~~~~~

../../../node_modules/@types/webpack/index.d.ts:25:19 - error TS2305: Module '"/Users/wujekbogdan/htdocs/node_modules/@types/tapable/index"' has no exported member 'HookMap'.

25 import { Tapable, HookMap,
                     ~~~~~~~

../../../node_modules/@types/webpack/index.d.ts:26:10 - error TS2305: Module '"/Users/wujekbogdan/htdocs/node_modules/@types/tapable/index"' has no exported member 'SyncBailHook'.

26          SyncBailHook, SyncHook, SyncLoopHook, SyncWaterfallHook,
            ~~~~~~~~~~~~

../../../node_modules/@types/webpack/index.d.ts:26:24 - error TS2305: Module '"/Users/wujekbogdan/htdocs/node_modules/@types/tapable/index"' has no exported member 'SyncHook'.

26          SyncBailHook, SyncHook, SyncLoopHook, SyncWaterfallHook,
                          ~~~~~~~~

../../../node_modules/@types/webpack/index.d.ts:26:34 - error TS2305: Module '"/Users/wujekbogdan/htdocs/node_modules/@types/tapable/index"' has no exported member 'SyncLoopHook'.

26          SyncBailHook, SyncHook, SyncLoopHook, SyncWaterfallHook,
                                    ~~~~~~~~~~~~

../../../node_modules/@types/webpack/index.d.ts:26:48 - error TS2305: Module '"/Users/wujekbogdan/htdocs/node_modules/@types/tapable/index"' has no exported member 'SyncWaterfallHook'.

26          SyncBailHook, SyncHook, SyncLoopHook, SyncWaterfallHook,
                                                  ~~~~~~~~~~~~~~~~~

../../../node_modules/@types/webpack/index.d.ts:27:10 - error TS2305: Module '"/Users/wujekbogdan/htdocs/node_modules/@types/tapable/index"' has no exported member 'AsyncParallelBailHook'.

27          AsyncParallelBailHook, AsyncParallelHook, AsyncSeriesBailHook, AsyncSeriesHook, AsyncSeriesWaterfallHook } from 'tapable';
            ~~~~~~~~~~~~~~~~~~~~~

../../../node_modules/@types/webpack/index.d.ts:27:33 - error TS2305: Module '"/Users/wujekbogdan/htdocs/node_modules/@types/tapable/index"' has no exported member 'AsyncParallelHook'.

27          AsyncParallelBailHook, AsyncParallelHook, AsyncSeriesBailHook, AsyncSeriesHook, AsyncSeriesWaterfallHook } from 'tapable';
                                   ~~~~~~~~~~~~~~~~~

../../../node_modules/@types/webpack/index.d.ts:27:52 - error TS2305: Module '"/Users/wujekbogdan/htdocs/node_modules/@types/tapable/index"' has no exported member 'AsyncSeriesBailHook'.

27          AsyncParallelBailHook, AsyncParallelHook, AsyncSeriesBailHook, AsyncSeriesHook, AsyncSeriesWaterfallHook } from 'tapable';
                                                      ~~~~~~~~~~~~~~~~~~~

../../../node_modules/@types/webpack/index.d.ts:27:73 - error TS2305: Module '"/Users/wujekbogdan/htdocs/node_modules/@types/tapable/index"' has no exported member 'AsyncSeriesHook'.

27          AsyncParallelBailHook, AsyncParallelHook, AsyncSeriesBailHook, AsyncSeriesHook, AsyncSeriesWaterfallHook } from 'tapable';
                                                                           ~~~~~~~~~~~~~~~

../../../node_modules/@types/webpack/index.d.ts:27:90 - error TS2305: Module '"/Users/wujekbogdan/htdocs/node_modules/@types/tapable/index"' has no exported member 'AsyncSeriesWaterfallHook'.

27          AsyncParallelBailHook, AsyncParallelHook, AsyncSeriesBailHook, AsyncSeriesHook, AsyncSeriesWaterfallHook } from 'tapable';
                                                                                            ~~~~~~~~~~~~~~~~~~~~~~~~

npm ERR! code ELIFECYCLE
npm ERR! errno 2
npm ERR! functions@ build: `tsc`
npm ERR! Exit status 2
npm ERR!
npm ERR! Failed at the functions@ build script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/wujekbogdan/.npm/_logs/2019-10-08T12_40_23_813Z-debug.log
error Command failed with exit code 2.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

I tried:
- Different node versions (8, 10, 12)
- Updating all depdnencies/devDependencies
- installing `@types/tapable` with `yarn addd @types/tapable --dev`

I'm not using `tabable` explicitly, it's a dependency of one of my dependencies.

---
The weird thing is that the project I'm trying to build used to work fine. I didn't change anything in the code since a half a year. Now I'm having these errors.

**Comment by evilebottnawi** [comment]
You don't need `@types/tapable`, we export types https://github.com/webpack/tapable/blob/master/tapable.d.ts

**Comment by wujekbogdan** [comment]
> You don't need @types/tapable, we export types

I installed it only as a workaround because I found a similar issue: https://github.com/webpack/tapable/issues/53

But this workaround doesn't work in my case.

**Comment by sokra** [comment]
You are using types from https://github.com/DefinitelyTyped/DefinitelyTyped
report issues with these types there.

**Comment by sokra** [comment]
@types/webpack seem to conflict with @types/tapable

probably you have mismatching versions of these

**Comment by blazzy** [comment]
Running `yarn why @types/webpack` and `yarn why @types/tapable` might provide you with some insight here. When I ran it I found that `clean-webpack-plugin` was pulling in those dependencies for me:

```
=> Found "clean-webpack-plugin#@types/webpack@4.41.25"
info This module exists because "_project_#browser#clean-webpack-plugin" depends on it.
```

I'm not sure why `@types/webpack` is a dependency and not a _dev_ dependency in [that project](https://github.com/johnagan/clean-webpack-plugin/blob/master/package.json#L80). If you're having the same issue, can find a slow/stalled discussion about it here: https://github.com/johnagan/clean-webpack-plugin/issues/127

Even if the definitely typed definitions are updated, clean-webpack-plugin would also need to be updated. The project hasn't had a release in 2 years, so I'm not holding out hope there. I ended up dropping the dependency and hacking together something simple to clear out my build folder.
